src/sdk.rs

```diff
@@ -328,6 +330,8 @@ pub fn self_deploy(code_key: &[u8]) {
         // Remove code from storage and store it in register 1.
         exports::storage_remove(code_key.len() as _, code_key.as_ptr() as _, 1);
         exports::promise_batch_action_deploy_contract(promise_id, u64::MAX as _, 1);
+        // TODO: Call upgrade on the same promise to make sure the state has migrated successfully.
```

Comment:
@evgenykuzyakov I don't understand what you mean here. Could you elaborate further?